core/src/main/java/io/questdb/griffin/SqlCompiler.java

```diff
@@ -1464,7 +1404,7 @@ private CompiledQuery insertAsSelect(ExecutionModel executionModel, SqlExecution
             final int cursorTimestampIndex = cursorMetadata.getTimestampIndex();
 
             // fail when target table requires chronological data and cursor cannot provide it
-            if (writerTimestampIndex > -1 && cursorTimestampIndex == -1) {
+            if (writerTimestampIndex > -1 && cursorTimestampIndex == -1 && cursorMetadata.getColumnType(writerTimestampIndex) != ColumnType.TIMESTAMP) {
```

Comment:
hmm, lets consider the following:

- user is inserting "string" into timestamp column? what do they get as an error?
- user chose not to populate timestamp column of writer? what do they get as an error?
- user assigned column X to timestamp column of the writer, but somehow nominated Y to be column of the cursor? what do they get as an error?